### DISCUSSION TITLE
Is rocketmq considering supporting HTTP Restful protocols?

### DISCUSSION THREAD

**Discussion by socutes** [apache/rocketmq#6304] [Ideas]
Rocketmq 5.0 has an independent proxy module and supports both grpc and pop protocols. Both kafka and pulsar support http restful production and consumption.
Supporting http restful in Proxy seems feasible.Whether the community has this plan and whether it needs to be discussed.
In my opinion, it is necessary in terms of ease of use and cost of development of the SDK.
I'd like to hear from the experts.

**Comment by echooymxq** [comment]
Yes, It is not difficult to integrate POP consumption with HTTP protocol in Proxy. But I want to know what are the use cases for using HTTP? If we only consider the ease of use of the SDK, the gRPC client in version 5.0 seems to be simpler.

**Reply by echooymxq** [reply]
> Yes, It is not difficult to integrate POP consumption with HTTP protocol in Proxy. But I want to know what are the use cases for using HTTP? If we only consider the ease of use of the SDK, the gRPC client in version 5.0 seems to be simpler.

Here is a [RIP-17](https://github.com/apache/rocketmq/wiki/RIP-17-RocketMQ-HTTP-Proxy-Support). As far as I know, the community's demand for HTTP protocol is not high.

**Comment by aaron-ai** [comment]
AFAIK, Pulsar provides [WebSocket API](https://pulsar.apache.org/docs/2.10.x/client-libraries-websocket/) to access, and Kafka provides [kafka-rest](https://github.com/confluentinc/kafka-rest) to make it possible to access by RESTful API, but this project is not belong to ASF. Both of Pulsar and Kafka doesn't provide the official client, Maybe because the RESTful API is out-of-box enough?

So I think it is a nice proposal to provide various way to access RocketMQ service, but we should discuss the specified protocol we choose and what organization should the related project belongs to.

**Comment by socutes** [comment]
Yes. kafka includes http restful support as a feature of a commercial release.The primary purpose of this requirement is to support more forms of access.
So two questions to discuss:
1. Whether there is a need for support
2. Where to support

**Comment by echooymxq** [comment]
In fact, i recently need to implement the HTTP protocol on the proxy. But just to be able more simply use the aliyun http sdk to connect rocketmq.
